src/main/java/org/testng/JarFileUtils.java

```diff
@@ -44,7 +51,11 @@
             if (!foundTestngXml) {
                 Utils.log("TestNG", 1,
                         "Couldn't find the " + xmlPathInJar + " in the jar file, running all the classes");
-                suites.add(XmlSuiteUtils.newXmlSuiteUsing(classes));
+                XmlSuite suite = XmlSuiteUtils.newXmlSuiteUsing(classes);
+                if (this.mode != null) {
```

Comment:
If mode cannot be null, no need to check it ;)

Reply:
Fixed